src/charm.py

```diff
@@ -45,6 +53,10 @@ def __init__(self, *args):
         self.framework.observe(self.on.remove, self.on_remove)
         self.framework.observe(self.on.update_status, self.on_update_status)
         self.framework.observe(self.on.upgrade_charm, self.on_upgrade_charm)
+        self.framework.observe(
+            self.remote_write_consumer.on.endpoints_changed,
+            self.handle_endpoints_changed,
```

Comment:
Maybe use a different method name here.  `remote_write_consumer_changed` Perhaps?

```suggestion
            self.remote_write_consumer_changed,
```